### PR TITLE
[Profiling] Allow wildcards for custom indices

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
Several profiling APIs allow to specify custom event indices with the `indices` parameter. So far this parameter has allowed only one index and also disallowed wildcards. With this commit we extend this parameter to support multiple indices and each index can use wildcards.

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
Note: I'm labeling this as non-issue as the respective API is exclusive to Kibana and it is not yet in use.

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java [hunk 1]
```diff
@@ -32,7 +35,7 @@
 /**
  * A request to get profiling details
  */
-public class GetStackTracesRequest extends ActionRequest implements IndicesRequest {
+public class GetStackTracesRequest extends ActionRequest implements IndicesRequest.Replaceable {
```
Comment (danielmitterdorfer) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java:38]: Implementing `IndicesRequest.Replaceable` indicates to infrastructure in Elasticsearch that we want to support wildcards, see also https://github.com/elastic/elasticsearch/blob/510c8515abecf5ca23154e25a72247329e8aa311/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java#L136-L146

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java [hunk 2]
```diff
@@ -221,6 +219,28 @@ public void parseXContent(XContentParser parser) throws IOException {
...
+                        + XContentParser.Token.VALUE_STRING
+                        + "] but found ["
+                        + token
+                        + "] in ["
+                        + INDICES_FIELD.getPreferredName()
+                        + "]."
+                );
+            }
+        }
+        return indices.toArray(new String[0]);
```
Comment (rockdaboot) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java:241]: Should the String array be pre-allocated with `indices.size()`?

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java [hunk 3]
```diff
@@ -330,7 +350,7 @@ public String[] indices() {
         if (this.indices == null) {
             indices.addAll(EventsIndex.indexNames());
         } else {
-            indices.add(this.indices);
+            indices.addAll(List.of(this.indices));
         }
         return indices.toArray(new String[0]);
```
Comment (rockdaboot) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java:355]: Maybe use pre-allocation here as well?

**Review by rockdaboot — APPROVED** [review]
LGTM (left some minor comments)

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java (hunk 2 above)
Comment (danielmitterdorfer) [reply x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java]: That's actually the preferred style in Java, see also https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html for the rationale.

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java (hunk 3 above)
Comment (danielmitterdorfer) [reply x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java:362]: That's actually the preferred style in Java, see also https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html for the rationale.

**Pull request merged**